staging/src/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go

```diff
@@ -231,7 +238,16 @@ func buildImpersonationRequests(headers http.Header) ([]v1.ObjectReference, erro
 		}
 	}
 
-	if (hasGroups || hasUserExtra) && !hasUser {
+	requestedUID := headers.Get(authenticationv1.ImpersonateUIDHeader)
+	hasUID := len(requestedUID) > 0
+	if hasUID {
+		impersonationRequests = append(impersonationRequests, v1.ObjectReference{
+			Kind: "UID",
```

Comment:
I am unsure if this should be:

```suggestion
			Kind: "Uid",
```

Reply:
We capitalize all of UID: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go#L501